### PR TITLE
CLOUDSTACK-9610: Disabled Host Keeps Being up status after unmanaging …

### PR DESCRIPTION
**Pull request by priyankparihar**
For complete description please refer -> [CLOUDSTACK-9605](https://issues.apache.org/jira/browse/CLOUDSTACK-9605)

### CODE REVIEW

**Comment by syed** [comment]
Hi @priyankparihar,

Thanks for the patch. Can you provide a more descriptive message about the bug and how this fix addresses that?

**Comment by rhtyd** [comment]
@priyankparihar can you add a marvin test for this, also change the PR's branch to 4.9?

**Comment by priyankparihar** [comment]
Hi @syed,
I hope modified description, answers your question.

**Comment by syed** [comment]
Thank you :+1:

**Comment by priyankparihar** [comment]
Hi Bhai Sahab @rhtyd,
Integration test 'test_04_disable_host_unmanage_cluster_check_hosts_status' is validating this behavior.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by priyankparihar** [comment]
@rhtyd and @syed. If everything looks good. Please provide your precious LGTM.

**Comment by karuturi** [comment]
@priyankparihar can you run internal CI and post @cloudmonger test results?

**Comment by priyankparihar** [comment]
>can you run internal CI and post @cloudmonger test results?

@karuturi running.

**Comment by rhtyd** [comment]
I think we'll need to discuss, I'm not sure if we should hide hosts that are disabled and cluster is un-managed. /cc @DaanHoogland  @abhinandanprateek comment?

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 408
 Hypervisor xenserver
 NetworkType Advanced
 Passed=104
 Failed=1
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Comment by DaanHoogland** [comment]
@priyankparihar I agree that host should not show as up but indeed why hide it? Unless we show no details for an unmanaged cluster we should be able to find the host in there. It should still show as disabled of course.

**Comment by priyankparihar** [comment]
Hi @DaanHoogland code is modified. Now we are not hiding. It will be shown in Disconnected state.

**Comment by DaanHoogland** [comment]
Looks good to me, @rhtyd are your concerns met?

**Comment by rhtyd** [comment]
LGTM, @abhinandanprateek ?

File: server/src/com/cloud/resource/ResourceManagerImpl.java [hunk 1]
```diff
@@ -2488,6 +2488,22 @@ public boolean maintenanceFailed(final long hostId) {
     }
 
     @Override
+    public List<HostVO> listAllHosts(Type type, Long clusterId, Long podId, long dcId) {
```
Comment (abhinandanprateek) [inline server/src/com/cloud/resource/ResourceManagerImpl.java]: make the input params final.

Comment (priyankparihar) [inline server/src/com/cloud/resource/ResourceManagerImpl.java]: Modified. Thank you for your careful observation.

**Comment by abhinandanprateek** [comment]
code review LGTM

**Comment by priyankparihar** [comment]
@abhinandanprateek ji,  If now everything LGTY then please provide your LGTM kind of blessing.

**Comment by priyankparihar** [comment]
tag:mergeready

PS: Hujur code ka tohfa kabool kijiye.

**Pull request merged**